src/adapter/src/coord/message_handler.rs

```diff
@@ -664,11 +667,17 @@ impl Coordinator {
                 None => return,
             };
 
-        // Re-validate that the catalog hasn't changed.
-        if transient_revision != self.catalog().transient_revision() {
-            // TODO(jkosh44) It would be preferable to re-validate the query instead of blindly failing.
```

Comment:
This is still probably some-what desirable.

Reply:
The new code does this though! It saves the very top level plan and re-issues it from the beginning. Or am I missing something?

Reply:
I think the intention of that comment and issue #17148, was that it would be possible and preferable to do more fine grained conflict checking other than just looking at the transient revision. For example, if some `CREATE TABLE` ran and it had no effect on my `SELECT`, then we'd continue the `SELECT` without re-planning/failing.

Reply:
Ah, yes, agree. Presumably we should have a general function to do that for all the various queries below you detail that could use it.